app/lib/domain_resource.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class DomainResource
+  attr_reader :domain
+
+  TIMEOUT_LIMIT = 5
```

Comment:
Nitpick, but something like `RESOLVE_TIMEOUT` would be a better name imho.

Reply:
Renamed constant.